src/dataregistry/registrar/dataset.py

```diff
@@ -417,7 +417,42 @@ def register(
                 name, kwargs_dict["version_string"], kwargs_dict["version_suffix"]
             )
 
-        # Make sure there is not already an entry with this name/version combination
+        # Make sure the relative_path in the `root_dir` is avaliable
+        if kwargs_dict["location_type"] in ["dataregistry", "dummy"]:
+            previous_datasets = self._find_previous(
+                None,
+                None,
+                None,
+                kwargs_dict["owner"],
+                kwargs_dict["owner_type"],
+                relative_path=kwargs_dict["relative_path"],
+            )
+
+            # If results are found, we can only use the old `relative_path` is
+            # that dataset is now deleted (and not archived)
+            if len(previous_datasets) > 0:
+                dest = _form_dataset_path(
+                    kwargs_dict["owner_type"],
+                    kwargs_dict["owner"],
+                    kwargs_dict["relative_path"],
+                    schema=self._schema,
+                    root_dir=self._root_dir,
+                )
+
+                if get_dataset_status(previous_datasets[-1].status, "archived"):
```

Comment:
I think you need to check all entries returned by _find_previous, not just the one that happens to be last when ordered by `replace_iteration`.   In fact, within `_find_previous` I would move the line adding the ordering to `stmt` to within the second part of the `if`.   It serves no purpose for the case where you're searching on `relative_path`

Reply:
Yes you are right, the `full_name` entry with the highest `replace_iteration` would always win.

I've changed the `relative_path` check to now order by `creation_date` so that its always checking against the latest entry using that `relative_path`.

I've also updated the test in `test_register_dataset_real_data.py` to make sure its comparing against the latest entry.

Reply:
Also I don't think there should ever be a scenario where `old_location` is set (i.e., we are ingesting data) and something already exists at the `relative_path`.

So I added a check during registration making sure nothing is at the destination location if `old_location` is set. And added a CI for it.

Let me know what you think

Reply:
That all sounds good.  I'll aim for re-reviewing today.